			     BASH PATCH REPORT
			     =================

Bash-Release:	5.1
Patch-ID:	bash51-013

Bug-Reported-by:	Anders Kaseorg <andersk@mit.edu>
Bug-Reference-ID:	<de65540274064d7b95f17e9c46a3a6d8@oc11expo7.exchange.mit.edu>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2021-11/msg00025.html

Bug-Description:

Bash did not always perform tilde expansion following an unquoted colon on
the rhs of an assignment statement in posix mode.

--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 12
+#define PATCHLEVEL 13
 
 #endif /* _PATCHLEVEL_H_ */
--- a/subst.c
+++ b/subst.c
@@ -3825,6 +3825,7 @@ expand_string_assignment (string, quoted
 #else
   td.flags = W_ASSIGNRHS;
 #endif
+  td.flags |= (W_NOGLOB|W_TILDEEXP);
   td.word = savestring (string);
   value = call_expand_word_internal (&td, quoted, 0, (int *)NULL, (int *)NULL);
   FREE (td.word);
